lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -421,7 +416,7 @@ void ThreadManager::Impl::removeWorker(size_t value) {
   {
     Synchronized s(workerMonitor_);
 
-    while (workerCount_ != workerMaxCount_) {
+    while (workerCount_ > goalCount) {
```

Comment:
It is not accessing `workerMaxCount_` outside of a lock. See line 424 where it uses the `Synchronized` object which grabs the mutex that the monitor owns.